### PR TITLE
Added new Q3D DC IR Example

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+edb.cutout(["1.2V_AVDLL_PLL", "1.2V_AVDDL", "1.2V_DVDDL"],
+           ["GND"],
+           output_aedb_path=output_edb,
+           use_pyaedt_extent_computing=True,
+           )
+
+
+###############################################################################
+# Identify pins position
+# ~~~~~~~~~~~~~~~~~~~~~~
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:44]:
```suggestion
# Identify pin positions
# ~~~~~~~~~~~~~~
```

Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:44]:
```suggestion
# Identify pin positions
# ~~~~~~~~~~~~~~
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+# Identify [x,y] pin locations on the components to define where to assign sources
+# and sinks for Q3D and append Z elevation.
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:46]:
```suggestion
# Identify [x,y] pin positions on the components to define where to assign sources
# and sinks for Q3D. Append the Z elevation.
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+pin_u11_scl = [i for i in edb.components["U11"].pins.values() if i.net_name == "1.2V_AVDLL_PLL"]
+pin_u9_1 = [i for i in edb.components["U9"].pins.values() if i.net_name == "1.2V_AVDDL"]
+pin_u9_2 = [i for i in edb.components["U9"].pins.values() if i.net_name == "1.2V_DVDDL"]
+
+
+###############################################################################
+# Append Z Positions
+# ~~~~~~~~~~~~~~~~~~
+# Note: The factor 100 converts from "meters" to "mm"
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:56]:
```suggestion
# Append Z positions
# ~~~~~~~~~~~~~
# Note: The factor 100 converts from "meters" to "mm"
```
Can the .. note:: directive be used in the comment?

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+location_u9_1_scl.append(edb.components["U9"].upper_elevation * 1000)
+
+location_u9_2_scl = [i * 1000 for i in pin_u9_2[0].position]
+location_u9_2_scl.append(edb.components["U9"].upper_elevation * 1000)
+
+
+###############################################################################
+# Identify 3D Components pins positions
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Identify the pin location where 3D components of passives will be added.
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:71]:
```suggestion
# Identify pin positions for 3D components
# ~~~~~~~~~~~~~~~~~~~~~~~~~~
# Identify the pin positions where 3D components of passives are to be added.
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+location_l2_1 = [i * 1000 for i in edb.components["L2"].pins["1"].position]
+location_l2_1.append(edb.components["L2"].upper_elevation * 1000)
+location_l4_1 = [i * 1000 for i in edb.components["L4"].pins["1"].position]
+location_l4_1.append(edb.components["L4"].upper_elevation * 1000)
+
+###############################################################################
+# Save and close Edb
+# ~~~~~~~~~~~~~~~~~~
+# Save, close Edb and open it in Hfss 3D Layout to generate the 3D model.
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:81]:
```suggestion
# Save and close EDB
# ~~~~~~~~~~~~
# Save and close EDB. Then, open EDT in HFSS 3D Layout to generate the 3D model.
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+edb.save_edb()
+edb.close_edb()
+
+h3d = pyaedt.Hfss3dLayout(output_edb, specified_version="2023.1", non_graphical=False, new_desktop_session=True)
+
+###############################################################################
+# Export to Q3D
+# ~~~~~~~~~~~~~
+# Create a dummy setup and export the layout in Q3D.
+# keep_net_name will reassign Q3D nets names from Hfss 3D Layout.
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:92]:
```suggestion
# The ``keep_net_name`` parameter reassigns Q3D net names from HFSS 3D Layout.
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+comp2.rotate(q3d.AXIS.Z, -90)
+comp2.parameters["n_turns"] = "3"
+comp2.parameters["d_wire"] = "100um"
+q3d.modeler.set_working_coordinate_system("Global")
+
+
+###############################################################################
+# Delete dielectrics
+# ~~~~~~~~~~~~~~~~~~
+#
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:133]: You want to have some type of description of what is happening in the code.

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+q3d.modeler.delete(q3d.modeler.get_objects_by_material("Megtron4"))
+q3d.modeler.delete(q3d.modeler.get_objects_by_material("Megtron4_2"))
+q3d.modeler.delete(q3d.modeler.get_objects_by_material("Megtron4_3"))
+q3d.modeler.delete(q3d.modeler.get_objects_by_material("Solder Resist"))
+
+###############################################################################
+# Assing Source and Sink
+# ~~~~~~~~~~~~~~~~~~~~~~
+# Use previously calculated position to identify faces and
+# assign sources and sinks on nets.
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:144]:
```suggestion
# Assign source and sink
# ~~~~~~~~~~~~~~~
# Use previously calculated positions to identify faces and
# assign sources and sinks on nets.
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+setup = q3d.create_setup()
+setup.dc_enabled = True
+setup.capacitance_enabled = False
+setup.ac_rl_enabled = False
+setup.props["SaveFields"] = True
+setup.props["DC"]["Cond"]["MaxPass"]=3
+setup.analyze()
+
+###############################################################################
+# Phi Plot
```
Comment (PipKat) [inline examples/05-Q3D/Q3D_DC_IR.py:175]:
```suggestion
# Phi plot
```

File: pyaedt/q3d.py [hunk 10]
```diff
@@ -1314,14 +1314,23 @@ def nets(self):
                 net_names.append(i[0].split(":")[1])
         return net_names
 
+    @pyaedt_function_handler()
+    def delete_all_nets(self):
+        """Delete all nets in current design."""
```
Comment (PipKat) [inline pyaedt/q3d.py:1319]:
```suggestion
        """Delete all nets in the design."""
```

File: examples/05-Q3D/Q3D_DC_IR.py [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+edb.cutout(["1.2V_AVDLL_PLL", "1.2V_AVDDL", "1.2V_DVDDL"],
+           ["GND"],
+           output_aedb_path=output_edb,
+           use_pyaedt_extent_computing=True,
+           )
+
+
+###############################################################################
+# Identify pin positions
+# ~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/05-Q3D/Q3D_DC_IR.py:45]:
```suggestion
# Identify pin positions
# ~~~~~~~~~~~~~~~~~~~~~~
```

**Pull request merged**